src/propmachinery.jl

```diff
@@ -20,6 +20,8 @@ QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix}(eq::QuLindbla
 
 QuPropagator{QPM<:QuPropagatorMethod, QM<:QuBase.AbstractQuMatrix, COT<:QuBase.AbstractQuMatrix}(hamiltonian::QuBase.AbstractQuMatrix, collapse_ops::Vector{COT}, init_state::QM, tlist, method::QPM) = QuPropagator{QPM,QM,QuLindbladMasterEq}(QuLindbladMasterEq(hamiltonian,collapse_ops), init_state, tlist, method)
 
+QuPropagator{QPM<:QuPropagatorMethod, QV<:QuBase.AbstractQuVector, COT<:QuBase.AbstractQuMatrix}(hamiltonian::QuBase.AbstractQuMatrix, collapse_ops::Vector{COT}, init_state::QV, tlist, method::QPM) = QuPropagator{QPM,QV,QuLindbladMasterEq}(QuLindbladMasterEq(hamiltonian,collapse_ops), init_state, tlist, method)
```

Comment:
One problem I wanted to point out is that with each `QuPropagator` we construct a `QuLindbladMasterEq(hamiltonian,collapse_ops)` which in turn creates and stores a Lindblad matrix. However, for MCWF we don't need this matrix and constructing it takes a lot of time. I don't have a satisfying solution yet. Maybe we equip `QuEquation` with a type parameter, which acts as a flag to indicate if the respective matrix is cached or not. Essentially, this is an issue independent of this PR, but I wanted to mention it so we don't forget it.